app/main/views/suppliers.py

```diff
@@ -143,3 +141,84 @@ def import_supplier(supplier_id):
         abort(400, "Database Error: {0}".format(e))
 
     return jsonify(suppliers=supplier.serialize()), 201
+
+
+@main.route('/suppliers/<int:supplier_id>', methods=['POST'])
+def update_supplier(supplier_id):
+    request_data = get_json_from_request()
+
+    supplier = Supplier.query.filter(
+        Supplier.supplier_id == supplier_id
+    ).first()
+
+    if supplier is None:
+        abort(404, "supplier_id '%d' not found" % supplier_id)
+
+    json_has_required_keys(request_data, ['suppliers'])
+
+    supplier_data = supplier.serialize()
+    supplier_data.update(request_data['suppliers'])
+    supplier_data = drop_foreign_fields(
+        supplier_data,
+        ['links', 'contactInformation']
+    )
+
+    validate_supplier_json_or_400(supplier_data)
+    json_has_matching_id(supplier_data, supplier_id)
+
+    supplier.update(supplier_data)
+
+    db.session.add(supplier)
+
+    try:
+        db.session.flush()
+        activity = Activity(verb=u'update', object=supplier)
+        db.session.add(activity)
+        db.session.commit()
+    except IntegrityError as e:
+        db.session.rollback()
+        abort(400, "Database Error: {0}".format(e))
+
+    return jsonify(suppliers=supplier.serialize())
+
+
+@main.route(
+    '/suppliers/<int:supplier_id>/contact-information/<int:contact_id>',
+    methods=['POST'])
+def update_contact_information(supplier_id, contact_id):
+    request_data = get_json_from_request()
+
+    contact = ContactInformation.query.filter(
+        ContactInformation.id == contact_id,
+        ContactInformation.supplier_id == supplier_id,
+    ).first()
+
+    if contact is None:
+        abort(404, "contact_id '%d' not found" % contact_id)
+
+    json_has_required_keys(request_data, ['contactInformation'])
+
+    contact_data = contact.serialize()
+    contact_data.update(request_data['contactInformation'])
+    contact_data = drop_foreign_fields(
+        contact_data,
+        ['links']
```

Comment:
Good call. I'm probably going to add `self` links for contact information objects in the supplier response, since otherwise the `/contact-infomation/` URLs aren't mentioned anywhere in the API responses.